From 5f39674dce8038f4a2f155a52b581fab0f3a3b0c Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Mon, 13 Dec 2021 12:16:26 +0800
Subject: [PATCH] mm/daemon: fix compile issue of damon/vaddr.c
Patch-mainline: Never, fix difference between openEuler and mainline code
openEuler-commit: 5b27dc83ddc5513414086eb9aed194884745f0ba
References: bsn#38

We need the header to resolve below errors.

mm/damon/vaddr.c: In function ‘damon_get_mm’:
mm/damon/vaddr.c:47:7: error: implicit declaration of function ‘get_task_mm’; did you mean ‘get_task_comm’? [-Werror=implicit-function-declaration]
  mm = get_task_mm(task);
       ^~~~~~~~~~~
       get_task_comm
mm/damon/vaddr.c:47:5: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  mm = get_task_mm(task);
     ^
mm/damon/vaddr.c: In function ‘damon_va_three_regions’:
mm/damon/vaddr.c:193:2: error: implicit declaration of function ‘mmput’; did you mean ‘fdput’? [-Werror=implicit-function-declaration]
  mmput(mm);
  ^~~~~
  fdput

Note, this is only part of commit 8581fd402a0c ("treewide: Add missing
includes masked by cgroup -> bpf dependency") because of conflict with
it's treewide change.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/damon/vaddr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/vaddr.c b/mm/damon/vaddr.c
index 35fe49080ee9..47f47f60440e 100644
--- a/mm/damon/vaddr.c
+++ b/mm/damon/vaddr.c
@@ -13,6 +13,7 @@
 #include <linux/mmu_notifier.h>
 #include <linux/page_idle.h>
 #include <linux/pagewalk.h>
+#include <linux/sched/mm.h>
 
 #include "prmtv-common.h"
 
-- 
2.26.2

